src/JET.jl

```diff
@@ -421,7 +421,15 @@ const StateAtPC = Tuple{State,Int}
 const LineTable = Union{Vector{Any},Vector{LineInfoNode}}
 
 get_stmt((sv, pc)::StateAtPC)         = @inbounds sv.src.code[pc]
-get_lin((sv, pc)::StateAtPC)          = @inbounds (sv.src.linetable::LineTable)[sv.src.codelocs[pc]]::LineInfoNode
+get_lin((sv, pc)::StateAtPC)          = begin
+    if 1 <= sv.src.codelocs[pc] <= length(sv.src.linetable::LineTable)
+        @inbounds (sv.src.linetable::LineTable)[sv.src.codelocs[pc]]::LineInfoNode
+    else
+        # Packages might dynamically generate code, which does not reference
+        # a source, see https://github.com/aviatesk/JET.jl/issues/273
+        LineInfoNode(sv.mod, :unknown, Symbol("unknown"), 0, 0)
+    end
```

Comment:
```suggestion
    codeloc = sv.src.codelocs[pc]
    linetable = sv.src.linetable::LineTable
    if 1 <= codeloc <= length(linetable)
        return @inbounds linetable[codeloc]::LineInfoNode
    else
        # Packages might dynamically generate code, which does not reference
        # a source, see https://github.com/aviatesk/JET.jl/issues/273
        return LineInfoNode(sv.mod, :unknown, :unknown, 0, 0)
    end
```
Minor clean ups and manual CSE.